src/performance.ts

```diff
@@ -0,0 +1,153 @@
+
+            const headers = {
+                'Content-Type': 'application/json',
+                'uuid': "94991193-4ca0-47ef-9637-729ea68a85de",
+            };
+            return fetch(url , {
+                method: 'post',
+                body:    JSON.stringify(reqData),
+                headers: headers,
+            });
+        } catch(err) {
+            alert(err);
+        }
+    };
+
+    getVulnerabilities = async (ecosystem, name, version) => {
+        try {
+            const part = [ecosystem, name, version].map(v => encodeURIComponent(v)).join('/');
+            let getUrl = "https://f8a-analytics-preview-2445582058137.staging.gw.apicast.io/api/v2/component-analyses/" + part + "?user_key=3e42fa66f65124e6b1266a23431e3d08";
+
+            return fetch(getUrl);
+        } catch(err) {
+            alert(err);
+        }
+    };
+
+    getBatchTime = async (ecosystem, payload) => {
+        let reqData = [];
+        for (let i = 0; i < payload.length; i += this.batchSize) {
+            reqData.push({
+                "ecosystem": ecosystem,
+                "package_versions": payload.slice(i, i + this.batchSize)
+            });
+        }
+        let batchStart = new Date().getTime();
+        const allRequests = reqData.map(request => this.fetchBatchVulnerabilities(request));
+        await Promise.allSettled(allRequests).then(() => {
+            let batchEnd = new Date().getTime();
+            let batchTime = batchEnd - batchStart;
+            console.log("\n Time taken for " + payload.length + " depensencies: [Batch call time = " + batchTime + " ms]");
+        });
+    }
+
+    getTime = async (ecosystem, payload) => {
+        let getStart = new Date().getTime();
+        const allRequests = payload.map(pack => this.getVulnerabilities(ecosystem, pack.package, pack.version));
+        await Promise.allSettled(allRequests).then(() => {
+            let getEnd = new Date().getTime();
+            let gTime = getEnd - getStart;
+            console.log("\n Time taken for " + payload.length + " depensencies: [Normal call time = " + gTime + " ms]");
+        });
+    }
+
+    performanceTest = async (ecosystem, payload) => {
+        const iterationCount = 10;
+        for (let i = 0; i < iterationCount; i++) {
+            console.log("\n iteration count:" + (i+1));
+            await this.getTime(ecosystem, payload);
+            setTimeout(() => {}, 2000);
+            await this.getBatchTime(ecosystem, payload);
+            setTimeout(() => {}, 2000);
+        }
+    }
+}
+
+async function runPerformance() {
+
+    let perf = new PerformanceComparision;
+
+    console.log("\n\n Comparision for 6 Dependencies: \n");
+    let payload6npm = [{"package":"lodash","version":"4.17.4"},{"package":"mongoose","version":"4.8.6"},{"package":"express-fileupload","version":"0.0.5"},{"package":"dustjs-linkedin","version":"2.5.0"},{"package":"adm-zip","version":"0.4.16"},{"package":"st","version":"2.0.0"}];
```

Comment:
Can these payloads read from file rather than embedded inside source?

Reply:
Having payload separate could also useful to run perf test using tools like ab(apache bench), wrk, jmeter